cloudinit/net/freebsd.py

```diff
@@ -0,0 +1,164 @@
+
+    def __init__(self, config=None):
+        if not config:
+            config = {}
+        self.dhcp_interfaces = []
+        self._postcmds = config.get('postcmds', True)
+
+    def _write_ifconfig_entries(self, settings, target=None):
+        ifname_by_mac = net.get_interfaces_by_mac()
+        for interface in settings.iter_interfaces():
+            device_name = interface.get("name")
+            device_mac = interface.get("mac_address")
+            if device_name and re.match(r'^lo\d+$', device_name):
+                continue
+            if device_mac and device_name:
+                cur_name = ifname_by_mac[device_mac]
+                if not cur_name:
+                    LOG.info('Cannot find any device with MAC %s', device_mac)
+                    continue
+                if cur_name != device_name:
+                    rhel_util.update_sysconfig_file(
+                        util.target_path(target, self.rc_conf_fn), {
+                            'ifconfig_%s_name' % cur_name: device_name})
+            elif device_mac:
+                device_name = ifname_by_mac[device_mac]
+
+            LOG.info('Configuring interface %s', device_name)
+            ifconfig = 'DHCP'  # default
+            for subnet in interface.get("subnets", []):
+
+                if subnet.get('type') == 'static':
+                    LOG.debug('Configuring dev %s with %s / %s', device_name,
+                              subnet.get('address'), subnet.get('netmask'))
+                # Configure an ipv4 address.
+                    ifconfig = (
+                            subnet.get('address') + ' netmask ' +
+                            subnet.get('netmask'))
+
+                # NOTE: Known limitation, we just set one subnet per interface
+                break
+
+            if ifconfig == 'DHCP':
+                self.dhcp_interfaces.append(device_name)
+            rhel_util.update_sysconfig_file(
+                util.target_path(target, self.rc_conf_fn), {
+                    'ifconfig_' + device_name: ifconfig})
+
+    def _write_route_entries(self, settings, target=None):
+        routes = list(settings.iter_routes())
+        for interface in settings.iter_interfaces():
+            subnets = interface.get("subnets", [])
+            for subnet in subnets:
+                if subnet.get('type') == 'static':
+                    gateway = subnet.get('gateway')
+                    if gateway:
+                        routes.append({
+                            'network': '0.0.0.0',
+                            'netmask': '0.0.0.0',
+                            'gateway': gateway})
+                    routes += subnet.get('routes', [])
+        route_cpt = 0
+        for route in routes:
+            network = route.get('network')
+            netmask = route.get('netmask')
+            gateway = route.get('gateway')
+            route_cmd = "-route %s/%s %s" % (network, netmask, gateway)
+            if not network:
```

Comment:
you can move this up to the first line so we skip any work if we know we can't do anything for this route in the loop.  Also, is this expected to have an incomplete route?  If not, let's log that we found a bad route and we are skipping it.